src/Microsoft.Unity.Analyzers.Tests/BaseTest.cs

```diff
@@ -70,13 +74,32 @@ public async Task DoNotFailWithInterfaceMembers()
 		await VerifyCSharpDiagnosticAsync(BaseDiagnosticVerifierTest<TAnalyzer>.InterfaceTest);
 	}
 
-	protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
+	protected override TAnalyzer GetCSharpDiagnosticAnalyzer()
 	{
 		return new TAnalyzer();
 	}
 
-	protected override CodeFixProvider GetCSharpCodeFixProvider()
+	protected override TCodeFix GetCSharpCodeFixProvider()
 	{
 		return new TCodeFix();
 	}
+
+	protected bool MethodExists(string typeName, string methodName, string assenblyNameFilter = "")
+	{
+		foreach (var assemblyFile in UnityAssemblies().Where(n => n.Contains(assenblyNameFilter)))
```

Comment:
the Where + Contains feels unnecessary right now as we're only using it for one case where we're looking for 1 assembly in particular. I'd just Select the one assembly we're looking for and look into that.